services/src/main/java/org/broadleafcommerce/payment/service/gateway/DefaultPayPalCheckoutTransactionService.java

```diff
@@ -96,77 +114,73 @@ public PaymentResponse authorize(PaymentRequest paymentRequest) throws PaymentEx
+        } catch (Exception e) {
+            processException(e, paymentResponse, paymentRequest, DefaultTransactionTypes.AUTHORIZE);
         }
         return paymentResponse;
     }
 
     @Override
     public PaymentResponse capture(PaymentRequest paymentRequest) throws PaymentException {
-        PaymentResponse responseDTO = new PaymentResponse(DefaultPaymentTypes.THIRD_PARTY_ACCOUNT,
-                PayPalCheckoutPaymentGatewayType.PAYPAL_CHECKOUT);
+        PaymentResponse paymentResponse =
+                new PaymentResponse(DefaultPaymentTypes.THIRD_PARTY_ACCOUNT,
+                        PayPalCheckoutPaymentGatewayType.PAYPAL_CHECKOUT);
 
         try {
             Authorization auth = getAuthorization(paymentRequest);
             Capture capture = capturePayment(paymentRequest, auth);
             Amount amount = capture.getAmount();
 
-            responseDTO
+            paymentResponse
                     .successful(true)
                     .rawResponse(capture.toJSON())
                     .transactionType(DefaultTransactionTypes.CAPTURE)
                     .responseMap(MessageConstants.CAPTUREID, capture.getId())
                     .amount(new BigDecimal(amount.getTotal()))
                     .currencyContext(amount.getCurrency());
-        } catch (PaymentException ex) {
-            if (ex.getCause() instanceof PayPalRESTException) {
-                PayPalRESTException restException = (PayPalRESTException) ex.getCause();
-                responseDTO
-                        .successful(false)
-                        .rawResponse(restException.toString())
-                        .transactionType(DefaultTransactionTypes.CAPTURE);
-                populateErrorResponseMap(responseDTO, restException);
-                return responseDTO;
-            }
-            throw ex;
+        } catch (PaymentException e) {
+            processException(e, paymentResponse, paymentRequest, DefaultTransactionTypes.CAPTURE);
         }
-        return responseDTO;
+        return paymentResponse;
     }
 
     @Override
     public PaymentResponse authorizeAndCapture(PaymentRequest paymentRequest)
             throws PaymentException {
-        PaymentResponse responseDTO = new PaymentResponse(DefaultPaymentTypes.THIRD_PARTY_ACCOUNT,
-                PayPalCheckoutPaymentGatewayType.PAYPAL_CHECKOUT);
+        PaymentResponse paymentResponse =
+                new PaymentResponse(DefaultPaymentTypes.THIRD_PARTY_ACCOUNT,
+                        PayPalCheckoutPaymentGatewayType.PAYPAL_CHECKOUT);
 
         try {
+            String transactionReferenceId = paymentRequest.getTransactionReferenceId();
+            if (StringUtils.isNotBlank(transactionReferenceId)) {
+                String paymentId =
+                        (String) paymentRequest.getAdditionalField(MessageConstants.PAYMENTID);
+
+                payPalPaymentService.updatePaymentCustom(paymentId, transactionReferenceId);
+            }
```

Comment:
This looks like duplicated code from the authorize() method, potential to refactor out.